src/ros_license_toolkit/checks.py

```diff
@@ -337,3 +337,28 @@ def _print_info(self, info_str, files_with_uncovered_licenses,
                     lambda x: x[0] in files_with_uncovered_licenses or (
                         x[0] in files_not_matched_by_any_license_tag),
                     package.found_files_w_licenses.items()))))
+
+
+class LicenseFilesReferencedCheck(Check):
+    """Check if all found License file have a reference in package.xml."""
+
+    def _check(self, package: Package):
+        not_covered_texts: Dict[str, str] = {}
+        for filename, license_text in package.found_license_texts.items():
+            # skipping all declarations above the package
+            if '../' in filename:
```

Comment:
This could lead to problems if the path for some reason contains an unnecessary `../`. For example, `a/b/../` is the same as `a/`. See https://stackoverflow.com/a/37095733/1493204 for a solution.